ocs_ci/utility/pagerduty.py

```diff
@@ -0,0 +1,332 @@
+    kubeconfig = os.getenv("KUBECONFIG")
+    cmd = (
+        f"oc create secret generic {constants.PAGERDUTY_SECRET} "
+        f"--from-literal=PAGERDUTY_KEY={integration_key} -n openshift-storage "
+        f"--kubeconfig {kubeconfig} --dry-run -o yaml"
+    )
+    secret_data = exec_cmd(cmd, secrets=[integration_key]).stdout
+    with tempfile.NamedTemporaryFile(
+        prefix=f"{constants.PAGERDUTY_SECRET}_"
+    ) as secret_file:
+        secret_file.write(secret_data)
+        secret_file.flush()
+        exec_cmd(f"oc apply --kubeconfig {kubeconfig} -f {secret_file.name}")
+    logger.info("New integration key was set.")
+
+
+def check_incident_list(summary, urgency, incidents):
+    """
+    Check list of incidents that there are incidents with requested label
+    in summary and specific urgency. If some incident is missing then this check
+    fails.
+
+    Args:
+        summary (str): String that is part of incident summary
+        urgency (str): Incident urgency
+        incidents (list): List of incidents to check
+
+    Returns:
+        list: List of incidents that match search requirements
+
+    """
+    logger.info(f"Looking for incidents with summary {summary} and urgency {urgency}")
+    target_incidents = [
+        incident
+        for incident in incidents
+        if (summary in incident.get("summary") and incident.get("urgency") == urgency)
+    ]
+    if target_incidents:
+        logger.info(f"Incidents with summary {summary} were found: {target_incidents}")
+    else:
+        logger.info(
+            f"No incidents with summary {summary} and urgency {urgency} were found"
+        )
+    return target_incidents
+
+
+class PagerDutyAPI(object):
+    """
+    This is wrapper class for PagerDuty API:
+    https://developer.pagerduty.com/api-reference/
+
+    In order to use the API, there must be set api_key. That is taken from
+    AUTH/pagerduty/api_key in config.
+
+    """
+
+    _token = None
+    _endpoint = "https://api.pagerduty.com"
+
+    def __init__(self, token=None):
+        """
+        Constructor for PagerDutyAPI class.
+        """
+        self._token = token or config.AUTH["pagerduty"]["api_key"]
+
+    def get(self, resource, payload=None):
+        """
+        Get resources from PagerDuty API.
+
+        Args:
+            resource (str): Represents part of uri that specifies given
+                resource
+            payload (dict): Provide parameters to GET API call.
+                e.g. for `incidents` resource this can be
+                {"service_ids[]": <id>, "since": "2021-07-15T00:00:00Z", "time_zone": "UTC"}
+
+        Returns:
+            dict: Response from PagerDuty api
+
+        """
+        pattern = f"/{resource}"
+        headers = {
+            "Authorization": f"Token {self._token}",
+            "Accept": "application/vnd.pagerduty+json;version=2",
+        }
+
+        logger.debug(f"GET {self._endpoint + pattern}")
+        logger.debug(f"headers={headers}")
```

Comment:
I would be very careful with logging, trying not to leak the api key. This applies to rest of this class.

Reply:
Headers logging was removed